lib/model/internal_link.dart

```diff
@@ -96,7 +96,13 @@ Uri narrowLink(PerAccountStore store, Narrow narrow, {int? nearMessageId}) {
     fragment.write('/near/$nearMessageId');
   }
 
-  return store.realmUrl.replace(fragment: fragment.toString());
+  Uri result = store.realmUrl.replace(fragment: fragment.toString());
+  if (result.path.isEmpty) {
+    // Always ensure that there is a '/' right after the hostname.
+    // A generated URL without '/' looks odd and does not linkify.
```

Comment:
```suggestion
    // A generated URL without '/' looks odd,
    // and if used in a Zulip message does not get automatically linkified.
```

The not getting linkified is a specific fact about what Zulip does with URLs it spots in the plain text of a message; it's not a general fact about URLs, and this function isn't specific to generating something to go into a Zulip message.